src/Period.php

```diff
@@ -509,6 +530,37 @@ protected function roundDate(DateTimeInterface $date, int $precision): DateTimeI
         );
     }
 
+    protected static function convertDateIntervalToPrecision(DateInterval $interval): int
```

Comment:
Not sure if the `Period` class should be concerned with this functionality. Perhaps it's better to convert `Precision` to an abstract class and add a "static factory" over there. What do you think?

(side note: this is where built in enums in PHP would be useful)

Reply:
Another sidenote: moving this logic to `Precision` would also mean we don't need to keep track of that `DEFAULT_PRECISION` constant in the `Period` class.

Reply:
I like it, although I would prefer a Value Object that we could then delegate even more

```php

public function __construct(/* ... */)
{
    $this->precision = $precisionMask
        ? Precision::fromConstant($precisionMask)
        : Precision::default()
    ;
    $this->dateInterval = $this->precision->toDateInterval();
}

protected function ensurePrecisionMatches(Period $period): void
{
    if (!$this->precision->equals($other->precision)) {
        throw CannotComparePeriods::precisionDoesNotMatch()
    }
}

public function precisionMask(): int
{
    return $this->precision->value();
}

// For devs that extended Period:
/**
  * @deprecated 1.5 Use $this->precision->toDateInterval() directly
  */
protected function createDateInterval(int $precision): DateInterval
{
    return $this->precision->toDateInterval();
}
```

I would love to do it, but then we would either have to retarget this PR at the 2.x release, or do the refactoring of precision handling in another PR.

Reply:
I think that's taking it a step too far for now. Are you ok with making `Precision` an abstract class, and reevaluate this for v2?

Reply:
I have a personal aversion against abstract classes (I don‘t mean to sound elitist, it‘s more emotional).

As we already have a mapping approach in  https://github.com/spatie/period/blob/83f21c1e391bb6b7e367f4a0d92aa04181ea8937/src/Period.php#L512 , perhaps we could do the same here as you proposed in the other review comment and revisit the general concept separately?